docs/contributing/merging-to-master-on-github.md

```diff
@@ -0,0 +1,41 @@
+# Merging to Master on Github
+
+We use several different approaches to merging commits into master. This article describes when to use which approach.
+All the scenarios presume that there are no conflicts with the base branch (i.e. with `master`). Otherwise, you have to resolve the conflicts locally. 
+
+**Scenario 1: I have only one commit in my branch and I want to merge it into `master` branch.**
+- I can use the "Squash and merge" button on PR detail.
+
+![Squash and merge](./img/github-squash-and-merge.png)
+
+**Scenario 2: I have multiple commits in my branch but I want to discard them and merge all the changes as one commit into `master` (i.e. I do not want to keep the history of my commits).**
+- same as scenario 1, I can use "Squash and merge".
+
+**Scenario 3: I have multiple commits in my branch and I want to merge them all into `master` using merge commit (i.e. I want to keep the history of my commits).**
+- When my branch is rebased on current `master`, I can use the "Merge pull request" button on PR detail (i.e. "Create a merge commit" option).
+I have to add some nice commit message that summarizes the whole modification and include the PR number in parentheses (e.g. `release of version 7.0.0-beta1 (#485)`).
```

Comment:
nice? i would ommit that

Reply:
What about "suitable"? Will be that better?

Reply:
Done